hawkular-alerts-engine/src/main/resources/org/hawkular/alerts/engine/rules/ConditionMatch.drl

```diff
@@ -378,7 +379,37 @@ rule AlertOnSatisfiedDampening
                 notifications.send(notification);
             }
         }
-        insert(  newAlert );
+
+        //insert(  newAlert );
+
         $d.reset();
         update( $d )
+
+        if ($t.isSafetyEnabled()) {
+            if (log != null && log.isDebugEnabled()) {
+                log.debug("Setting Trigger to Safety Mode! " + $t);
+            }
+            $t.setMode(Mode.SAFETY);
+            update( $t )
+
+        } else if (log != null && log.isDebugEnabled()) {
+            log.debug("Trigger remains in Fire mode, Safety Mode not enabled. " + $t);
+        }
+end
+
+
+rule SetFiringModeOnSatisfiedDampening
+    when
+        $t  : Trigger( mode == Mode.SAFETY, $tid : id )
+        $d  : Dampening( triggerMode == Mode.SAFETY, triggerId == $tid, satisfied == true ) 
+    then
+        if (log != null && log.isDebugEnabled()) {
+            log.debug("SetFiringModeOnSatisfiedDampening! " + $d.log());
+        }
+
+        $d.reset();
+        update( $d )
+
+        $t.setMode(Mode.FIRE);
+        update( $t )

```

Comment:
Same idea like the previous one: as we can modify the trigger from the UI, if the engine changes the status this should be updated via DefinitionsService to have a consistency between UI and persistence.
If not we can have the risk to have some "detached objects" in the session engine from the DefinitinosService.